### PR TITLE
perf: lazy trade routes

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR:

- makes trade routes components lazy
- doesn't fire snap installation checks unless we're in the context of a connected wallet, which isn't demo
- doesn't render the trade component unless n the context of a connected wallet, **or** demo wallet

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

- Snap installation checks may be broken
- Swapper may not render when it should
- Swapper may be borked

## Testing

- Do a smoke test of MM snap installation checks
- Do a smoke test of swapper
- Ensure swapper doesn't render until wallet is connected (i.e shouldn't render on locked MM, nor when the wallet is in enter password state)

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

- Native password - no swapper

image

- Native unlocked - swapper happy

image

- Locked MM - no swapper

image

- Unlocked MM - swapper happy

image

- Snap installed

image

- Snap uninstalled while MM is connected

image

- Snap installation check after refresh

image

- Snap installation check on a fresh connect

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5513** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5513" target="_blank">Graphite</a> 
    * **PR #5517** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5517" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5517?utm_source=stack-comment).

File: src/hooks/useIsSnapInstalled/useIsSnapInstalled.tsx [hunk 1]
```diff
@@ -39,16 +39,17 @@ export const useIsSnapInstalled = (): null | boolean => {
   const [isSnapInstalled, setIsSnapInstalled] = useState<null | boolean>(null)
 
   const {
-    state: { wallet },
+    state: { wallet, isConnected, isDemoWallet },
   } = useWallet()
 
   const checkSnapInstallation = useCallback(async () => {
+    if (!isConnected || isDemoWallet) return
```
Comment (gomesalexandre) [inline src/hooks/useIsSnapInstalled/useIsSnapInstalled.tsx:46]: While at it - additional safety not to spam JSON-RPC requests, and not even begin the `checkIsMetaMask` checks until we have a connected wallet

File: src/pages/Dashboard/DashboardSidebar.tsx [hunk 2]
```diff
@@ -57,6 +58,11 @@ const promoData: PromoItem[] = [
 const display = { base: 'none', xl: 'block' }
 
 export const DashboardSidebar = memo(() => {
+  const {
+    state: { isConnected, isDemoWallet },
+  } = useWallet()
+  if (!isConnected && !isDemoWallet) return null
```
Comment (gomesalexandre) [inline src/pages/Dashboard/DashboardSidebar.tsx:64]: To be confirmed with @reallybeard if we want this

Comment (gomesalexandre) [reply src/pages/Dashboard/DashboardSidebar.tsx:64]: Got that castor oil approval for beard growth

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**